redbiom/admin.py

```diff
@@ -239,70 +233,69 @@ def load_sample_metadata(metadata):
-        acquired = get('metadata', 'SETNX', '__load_md_lock/1') == 1
-        if not acquired:
-            # TODO: support verbose
-            time.sleep(1)
+    if tag is not None:
+        original_ids = md[md.columns[0]][:]
 
-    try:
-        # subset to only the novel IDs
-        represented = get('metadata', 'SMEMBERS', 'samples-represented')
-        md = md.loc[set(md.index) - set(represented)]
-        if len(md) == 0:
-            raise ValueError("No new sample IDs found in: %s" % metadata)
-
-        samples = md.index
-        indexed_columns = md.columns
-        for idx, row in md.iterrows():
-            # denote what columns contain information
-            columns = [c for c, i in zip(md.columns, row.values)
-                       if _indexable(i, null_values)]
-            key = "categories:%s" % idx
-
-            # TODO: express metadata-categories using redis sets
-            # TODO: dumps is expensive relative to just, say, '\t'.join
-            put('metadata', 'SET', key, json.dumps(columns))
-
-        for col in indexed_columns:
-            bulk_set = ["%s/%s" % (idx, v) for idx, v in zip(md.index, md[col])
-                        if _indexable(v, null_values)]
-
-            payload = "category:%s/%s" % (col, '/'.join(bulk_set))
-            post('metadata', 'HMSET', payload)
-
-        payload = "samples-represented/%s" % '/'.join(md.index)
-        post('metadata', 'SADD', payload)
-
-        payload = "categories-represented/%s" % '/'.join(md.columns)
-        post('metadata', 'SADD', payload)
-    except:
-        raise
-    finally:
-        # release the lock no matter what
-        get('metadata', 'DEL', '__load_md_lock')
+        # if the metadata are tagged, they must have sample metadata already
+        # loaded
+        if not redbiom.util.has_sample_metadata(original_ids):
+            raise ValueError("Sample metadata must be loaded first.")
+
+        # tag the sample IDs
+        md[md.columns[0]] = ['%s_%s' % (tag, i) for i in md[md.columns[0]]]
+
+    md.set_index(md.columns[0], inplace=True)
+
+    # subset to only the novel IDs
+    represented = get('metadata', 'SMEMBERS', 'samples-represented')
+    md = md.loc[set(md.index) - set(represented)]
+    if len(md) == 0:
+        return 0
+
+    samples = md.index
+    indexed_columns = md.columns
+    for idx, row in md.iterrows():
+        # denote what columns contain information
+        columns = [c for c, i in zip(md.columns, row.values)
+                   if _indexable(i, null_values)]
+        key = "categories:%s" % idx
+
+        # TODO: express metadata-categories using redis sets
+        # TODO: dumps is expensive relative to just, say, '\t'.join
+        put('metadata', 'SET', key, json.dumps(columns))
+
+    for col in indexed_columns:
+        bulk_set = ["%s/%s" % (idx, v) for idx, v in zip(md.index, md[col])
+                    if _indexable(v, null_values)]
+
+        payload = "category:%s/%s" % (col, '/'.join(bulk_set))
```

Comment:
Do you think it's worth changing "category:%s/%s" to: "category:%s:%s"? Wondering if it could lead to confusions as it looks like a relative path ...